.github/workflows/docs_workflow.yml

```diff
@@ -0,0 +1,95 @@
+name: CI/CD Release Workflow
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+
+  release:
+    types: [published]
+
+jobs:
+  publish-docs:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Cache Conda
+        uses: actions/cache@v4
+        env:
+          # Increase this value to reset cache if deploy/conda-dev-spec.template has not changed in the workflow
+          CACHE_NUMBER: 0
+        with:
+          path: ~/conda_pkgs_dir
+          key: ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{
+            hashFiles('dev-spec.txt') }}
+
+      - if: ${{ steps.skip_check.outputs.should_skip != 'true' }}
+        name: Set up Conda Environment
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          activate-environment: "mpas_analysis_ci"
+          miniforge-version: latest
+          channels: conda-forge
+          channel-priority: strict
+          auto-update-conda: true
+          python-version: ${{ matrix.python-version }}
+
+      - if: ${{ steps.skip_check.outputs.should_skip != 'true' }}
+        name: Install mpas_analysis
+        run: |
+          git config --global url."https://github.com/".insteadOf "git@github.com:"
+          conda create -n mpas_analysis_dev --file dev-spec.txt \
+            python=${{ matrix.python-version }}
+          conda activate mpas_analysis_dev
+          python -m pip install -vv --no-deps --no-build-isolation -e .
+
+      - name: Build Sphinx Docs
+        run: |
+          set -e
+          conda activate mpas_analysis_dev
+          pip check
+          mpas_analysis sync diags --help
+          cd docs
+          sphinx-multiversion . _build/html
+      - name: Copy Docs and Commit
+        run: |
+          set -e
+          conda activate mpas_analysis_dev
+          pip check
+          mpas_analysis sync diags --help
+          cd docs
+          # gh-pages branch must already exist
+          git clone https://github.com/MPAS-Dev/MPAS-Analysis.git --branch gh-pages --single-branch gh-pages
+          # Make sure we're in the gh-pages directory.
+          cd gh-pages
+          # Create `.nojekyll` (if it doesn't already exist) for proper GH Pages configuration.
+          touch .nojekyll
+          # Add `index.html` to point to the `develop` branch automatically.
+          printf '<meta http-equiv="refresh" content="0; url=./develop/index.html" />' > index.html
+          # Only replace `develop` docs with latest changes. Docs for releases should be untouched.
+          rm -rf develop
```

Comment:
@altheaden, as we discussed, let's make sure `develop` here becomes the name of the target branch.  See if you can figure that out based on the link I sent you today. If not, no worries.  We can look it up together.  The redirect should still be for `develop` regardless of the target branch, though.